compiler/mir/rtchecks.nim

```diff
@@ -397,6 +397,131 @@ proc emitObjectCheck(tree; call; graph; env; bu) =
     bu.emitCall(tree, call, env.addCompilerProc(graph, "raiseObjectConversionError")):
       discard
 
+# XXX: currently cannot be moved to within ``emitCheckedBinaryIntOp`` due to a
+#      csource compiler bug
+const ArithChcks: array[mAddI..mModI, array[tyInt..tyInt64, string]] = block:
+  # compile the names of the procedure at compile-time
+  var res: array[mAddI..mModI, array[tyInt..tyInt64, string]]
+  for op, it in res.mpairs:
+    let base = case op
+      of mAddI: "nimAddInt"
+      of mSubI: "nimSubInt"
+      of mMulI: "nimMulInt"
+      of mDivI: "nimDivInt"
+      of mModI: "nimModInt"
+
+    # the inner array stores the per-width names, which is the base name
+    # suffixed with the width
+    for i in 0..3:
+      it[[tyInt8, tyInt16, tyInt32, tyInt64][i]] = base & $(8 shl i)
```

Comment:
```suggestion
    const intKinds = [tyInt8, tyInt16, tyInt32, tyInt64]
    for i, k in intKinds.pairs:
      it[k] = base & $(8 shl i)
```

I believe this is equivalent and a tad clearer, but not a required change.

Reply:
Yep, it's much easier to read.